### DISCUSSION TITLE
node browser modules expected?

### DISCUSSION THREAD

**Discussion by greggb** [vercel/next.js#16785] [Help] [answered]
I'm trying to audit the bundle sizes in my project, and I can't explain where some of these modules are coming from. Are they required by next.js in general, or am I triggering their inclusion somehow when using `next/link` or `next/router`? I'd be interested in trying an alternative approach in my code if there's something I'm doing to include these large-ish packages.

The big standouts for me are

- stream-browserify
- buffer
- readable-stream
- html-tokenize

<img width="765" alt="bundlesize" src="https://user-images.githubusercontent.com/514026/91995457-d86f1780-ed05-11ea-8964-7ccb80c2010c.png">

**Comment by timneutkens** [comment, marked as answer]
Those are all included through some npm library you're importing, they're not used in Next.js.

**Comment by greggb** [comment]
Thanks @timneutkens Any idea why I'd see this output for `yarn why`?
```
➜ yarn why stream-browserify
yarn why v1.22.5
[1/4] 🤔  Why do we have the module "stream-browserify"...?
[2/4] 🚚  Initialising dependency graph...
[3/4] 🔍  Finding dependency...
[4/4] 🚡  Calculating file sizes...
=> Found "stream-browserify@3.0.0"
info Reasons this module exists
   - "next" depends on it
   - Hoisted from "next#stream-browserify"
info Disk size without dependencies: "292KB"
info Disk size with unique dependencies: "532KB"
info Disk size with transitive dependencies: "772KB"
info Number of shared dependencies: 7
✨  Done in 0.58s.
```

**Reply by timneutkens** [reply]
Because it's part of the `stream` polyfilling. It's only included when `stream` is used by a dependency
